go-controller/pkg/ovn/pods.go

```diff
@@ -337,15 +335,24 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 	}
 
 	if annotation == nil {
-		gwIfAddr := util.GetNodeGatewayIfAddr(nodeSubnet)
-		routes, gwIP, err := getRoutesGatewayIP(pod, gwIfAddr)
-		if err != nil {
-			return err
-		}
-
 		var gwIPs []net.IP
-		if gwIP != nil {
-			gwIPs = []net.IP{gwIP}
+		var routes []util.PodRoute
+		for _, podIfAddr := range podIfAddrs {
+			nodeSubnet, err := util.MatchIpFamily(utilnet.IsIPv6CIDR(podIfAddr), nodeSubnets)
+			if err != nil {
+				return err
+			}
+
+			var gwIP net.IP
+			gwIfAddr := util.GetNodeGatewayIfAddr(nodeSubnet)
+			routes, gwIP, err = getRoutesGatewayIP(routes, pod, gwIfAddr)
```

Comment:
It's weird that `routes` gets passed in, modified, and passed out, but `gwIPs` doesn't work the same way.

I'd suggested in #1142 that we could construct a partial `util.PodAnnotation` first, and then pass it to `getRoutesGatewayIP` to be modified (which would probably mean renaming it to `addRoutesGatewayIP` or something).

Reply:
Yes, I saw you note in #1142. I was trying to keep the amount of changes in each commit small and was going to follow up with that fix. I totally forgot to add a comment to code indicating that. Went ahead and added the suggested change now.

Done.